packages/material-ui/src/RadioGroup/RadioGroup.test.js

```diff
@@ -257,6 +260,61 @@ describe('<RadioGroup />', () => {
     });
   });
 
+  describe('useRadioGroup', () => {
+    const RadioGroupController = React.forwardRef((_, ref) => {
+      const radioGroup = useRadioGroup();
+      React.useImperativeHandle(ref, () => radioGroup, [radioGroup]);
+      return null;
+    });
+
+    const RadioGroupControlled = React.forwardRef(function RadioGroupControlled(props, ref) {
+      return (
+        <RadioGroup {...props}>
+          <RadioGroupController ref={ref} />
+        </RadioGroup>
+      );
+    });
+
+    describe('from props', () => {
+      it('should have the name prop from the instance', () => {
+        const radioGroupRef = React.createRef();
+        const { setProps } = render(<RadioGroupControlled name="group" ref={radioGroupRef} />);
+
+        expect(radioGroupRef.current).to.have.property('name', 'group');
+
+        setProps({ name: 'anotherGroup' });
+        expect(radioGroupRef.current).to.have.property('name', 'anotherGroup');
+      });
+
+      it('should have the value prop from the instance', () => {
+        const radioGroupRef = React.createRef();
+        const { setProps } = render(<RadioGroupControlled ref={radioGroupRef} value="" />);
+
+        expect(radioGroupRef.current).to.have.property('value', '');
+
+        setProps({ value: 'one' });
+        expect(radioGroupRef.current).to.have.property('value', 'one');
+      });
+    });
+
+    describe('callbacks', () => {
+      describe('onChange', () => {
+        it('should set the value state', () => {
+          const radioGroupRef = React.createRef();
+          render(<RadioGroupControlled ref={radioGroupRef} defaultValue="zero" />);
```

Comment:
Not actually sure what `*Controlled` refers to in the original tests (whether it's to do with controlled/uncontrolled mode, or whether it's to do with `Controller`), but I guess in this case the component is uncontrolled, and I'm checking its state changes.

Reply:
Just bad naming I guess. I think I would now call it `RadioGroupHookListener` (or use the hooks testing library) and don't name the `RadioGroupControlled` at all.

Reply:
I can leave this as is, unless you have objections.

Reply:
It's fine. I'll revisit this at some point and give it better names across both tests.